source/blog/2015-10-31-rspec-compsable-matchers.md

```diff
@@ -0,0 +1,220 @@
+---
+title: How the RSpec 3.0 composable matchers work
+author: Sam Phippen
+---
+
+##Warmup: the `===` protocol
+
+In Ruby, a number of objects implement the `===` protocol. Some of the objects
+that implement this protocol include:
+
+* all class objects
+* lambdas
+* regexes
+* ranges
+
+Conceptually, this protocol represents a "match", but can be thought of as
+categorising instances of objects. That is, the object that implements `===`
+defines whether or not the passed in object belongs in the same category.
+For example, `Array === []` returns `true`. This is because an empty array is
+an instance of the array class. `Array === {}` returns false because a hash is
+not an array. For regexes, `===` returns true if the passed string matches the
+regex, and false if it does not.
+
+The `===` method is used in Ruby case statements to decide if a branch should
+be taken. For example
+
+```Ruby
+case 3
+when (1...18)
+  puts "hello"
+end
+```
```

Comment:
This isn't rendering right:

<img width="525" alt="screen shot 2016-01-19 at 11 33 19 pm" src="https://cloud.githubusercontent.com/assets/49391/12442469/1252fa68-bf05-11e5-8d56-45720f93ae12.png">